Split/Matchers/Semver/Semver.swift

```diff
@@ -0,0 +1,201 @@
+//
+//  Semver.swift
+//  Split
+//
+//  Copyright © 2024 Split. All rights reserved.
+//
+
+import Foundation
+
+enum SemverParseError: Error {
+    case invalidVersionFormat(String)
+}
+
+class Semver: Equatable {
+    private let kMetadataDelimiter: Character = "+"
+    private let kPreReleaseDelimiter: Character = "-"
+    private let kValueDelimiter: Character = "."
+    private var major: Int64
+    private var minor: Int64
+    private var patch: Int64
+    private var preRelease: [String] = []
+    private var isStable: Bool = true
+    private var metadata: String?
+
+    private var version: String = ""
+
+    static func build(version: String) -> Semver? {
+        return try? Semver(version)
+    }
+
+    private init(_ version: String) throws {
+        // set and remove metadata if exists
+        var vWithoutMetadata: String = ""
+        if let index = version.firstIndex(of: kMetadataDelimiter) {
+            let metadataIndex = version.index(after: index)
+
+            guard metadataIndex < version.endIndex else {
```

Comment:
I wouldn't use guard when no unwrapping an optional, because I think is confusing. But this is a personal opinion.